.github/workflows/makefile.yml

```diff
@@ -0,0 +1,37 @@
+name: Makefile CI
+
+on:
+  push:
+    branches: [ "main" ]
```

Comment:
What does this give us?

Reply:
It kind of just came in the template, but I think it will be useful to keep - this means when we merge to `main`, this will be re-run. This is mostly useful if there are a few open PRs, which means that each of those PRs did not test compatibility with the changes from the other PRs. Re-running that on pushes to `main` means that those changes will get re-tested against each other.

Reply:
Or, in other words -

- Someone submits PR `foo`, CI build runs + passes against current `main`
- Shortly afterwards, someone submits PR `bar`, and the build _also_ passes against `main`
- `foo` is merged, build still passes
- `bar` is merged, but is incompatible with changes from `foo`, and the build breaks (and, we'll know right away, hopefully)

Reply:
I expect the workflow to be rerun when the base branch is updated, but it is really hard to get a straight answer from Github on this.

> By default, a workflow only runs when a pull_request_target event's activity type is opened, synchronize, or reopened.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

So when is the activity type `synchronize`? I found nothing in the regular documentation, but the enterprise documentation has,

> synchronize: Triggered when a pull request's head branch is updated. For example, when the head branch is updated from the base branch, when new commits are pushed to the head branch, or when the base branch is changed.

https://docs.github.com/en/enterprise-server@3.6/developers/webhooks-and-events/webhooks/webhook-events-and-payloads

Reply:
Ah, I see - yeah, if GitHub takes care of re-running the CI when another PR gets merged, then that would remove the need for this (assuming, of course, we don't allow direct pushes to `main`, haha)! I could remove this and we could do a quick test after merging this PR

Reply:
Sweet. Let's do it.

Reply:
:shipit: